staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -102,15 +103,12 @@ func NewAvailableConditionController(
 
 	// construct an http client that will ignore TLS verification (if someone owns the network and messes with your status
 	// that's not so bad) and sets a very short timeout.
-	discoveryClient := &http.Client{
-		Transport: &http.Transport{
```

Comment:
a more contained fix would be to call `utilnet.SetTransportDefaults` on this transport, and turn off keep-alive, so we avoid hung blackholed connections

Reply:
If i read the code correctly, the transport which gets instantiated here does never actually get used.
The passed in `proxyTransport` is never nil. 

Modifying the passed in `proxyTransport` cannot be considered "safe"?